core/math/rect2.h

```diff
@@ -382,16 +382,16 @@ struct _NO_DISCARD_ Rect2i {
 			ERR_PRINT("Rect2i size is negative, this is not supported. Use Rect2i.abs() to get a Rect2i with a positive size.");
 		}
 #endif
-		if (position.x > (p_rect.position.x + p_rect.size.width)) {
+		if (position.x >= (p_rect.position.x + p_rect.size.width)) {
```

Comment:
Might be better to add an extra argument instead, like it was done for `Rect2` in https://github.com/godotengine/godot/pull/36021, and have both options.

Reply:
This would probably be better suited as a separate pull request.